File: sdk/julia/src/network_services.jl
# Network Services: IP addresses and reverse DNS for services
struct NetworkServicesResource
    http::HttpClient
end

# GET /api/service/{id}/ip - List Service IP Addresses
function list_ips(r::NetworkServicesResource; kwargs...)
    params = Dict(kwargs)
    http_get(r.http, "/api/service/{id}/ip"; params = params)
end

# GET /api/service/{id}/rdns - Get reverse DNS entries
function get_rdns(r::NetworkServicesResource; kwargs...)
    params = Dict(kwargs)
    http_get(r.http, "/api/service/{id}/rdns"; params = params)
end

# POST /api/service/{id}/rdns - Update reverse DNS entries
function update_rdns(r::NetworkServicesResource; kwargs...)
    params = Dict(kwargs)
    http_post(r.http, "/api/service/{id}/rdns"; params = params)
end

File: sdk/rust/src/network_services.rs
//! Network Services: IP addresses and reverse DNS for services

use std::sync::Arc;
use crate::http_client::HttpClient;

pub struct NetworkServicesResource {
    http: Arc<HttpClient>,
}

impl NetworkServicesResource {
    pub fn new(http: Arc<HttpClient>) -> Self { Self { http } }

    /// GET /api/service/{id}/ip - List Service IP Addresses
    pub async fn list_ips(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.get("/api/service/{id}/ip", None).await
    }

    /// GET /api/service/{id}/rdns - Get reverse DNS entries
    pub async fn get_rdns(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.get("/api/service/{id}/rdns", None).await
    }

    /// POST /api/service/{id}/rdns - Update reverse DNS entries
    pub async fn update_rdns(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.post("/api/service/{id}/rdns", None).await
    }

}
